### ISSUE TITLE
wrong `dmd --version` when building from git master: shows last tag instead of new one

### ISSUE THREAD

**Issue by dlangBugzillaToGithub** [dlang/dmd#19385]
### CC List

* Timothee Cour

### Description

````markdown
git rev-parse HEAD
7e8bea0e83314f39b8c8b2bddbfbff2b0bdee74d


dmd --version
DMD v2.078.2-beta.1-653-g7e8bea0e8 DEBUG
DMD64 D Compiler v2.078.2-beta.1-653-g7e8bea0e8
Copyright (C) 1999-2018 by The D Language Foundation, All Rights Reserved written by Walter Bright

should show instead the next version, ie something like:
v2.079-653-g7e8bea0e8 DEBUG

this is confusing in bug reports (eg https://issues.dlang.org/show_bug.cgi?id=18393 where dmd --version showed v2.078.2-beta.1-653-g7e8bea0e8 even though bug affects master, not 2.078.2-beta.1)

basically as soon as one creates a release tag (eg v2.078) the version identifier __VERSION__ should be bumped in git master
````
